cmd/api/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"github.com/getsentry/sentry-go/gin"
+	"github.com/gin-gonic/gin"
+	"github.com/spf13/viper"
+	"github.com/swaggo/gin-swagger"
+	"github.com/swaggo/gin-swagger/swaggerFiles"
+	"github.com/trustwallet/blockatlas/api"
+	"github.com/trustwallet/blockatlas/config"
+	_ "github.com/trustwallet/blockatlas/docs"
+	"github.com/trustwallet/blockatlas/pkg/ginutils"
+	"github.com/trustwallet/blockatlas/pkg/logger"
+	"github.com/trustwallet/blockatlas/platform"
+	"github.com/trustwallet/blockatlas/storage"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+)
+
+const (
+	defaultPort       = "8420"
+	defaultConfigPath = "../../config.yml"
+)
+
+var (
+	port, confPath string
+	cache          *storage.Storage
+	sg             gin.HandlerFunc
+)
+
+func init() {
+	sg = sentrygin.New(sentrygin.Options{})
+	cache = storage.New()
+
+	flag.StringVar(&port, "p", defaultPort, "port for api")
+	flag.StringVar(&confPath, "c", defaultConfigPath, "config file for api")
+
+	flag.Parse()
+
+	confPath, err := filepath.Abs(defaultConfigPath)
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	config.LoadConfig(confPath)
+	logger.InitLogger()
+	platform.Init()
+
+	host := viper.GetString("storage.redis")
+	err = cache.Init(host)
+	if err != nil {
+		logger.Fatal(err)
+	}
+}
+
+func main() {
```

Comment:
a method with almost 100 lines

Reply:
So what?

Reply:
Here are the example of large method in current architecture: https://github.com/trustwallet/blockatlas/blob/master/cmd/observer.go#L21

Reply:
yes, this is a good point to refactor and have better clean code, not the cmd package

Reply:
Current architecture is not using any good design pattern. If you could provide examples of projects that are using the same principles as Blockatlas - it will be great. We provide several examples of industry leading projects that have the same design pattern as we use in this PR. We have already explained that it will reduce the entry barrier for new developers as well as help to maintain project. Using complex framework for small issue is not a advantage of current solution as well